### PR TITLE
add `_is_k8s_node` label to the host labels

### PR DESCRIPTION
**Pull request by ilyam8**
##### Summary

Fixes netdata/product#1715

This PR adds whether the node is k8s node detection. This info added as a host label (`_is_k8s_node`).

##### Component Name

`daemon/`
`database/`
`web/`

##### Test Plan

tbd

### CODE REVIEW

File: web/api/netdata-swagger.json [hunk 1]
```diff
@@ -9,7 +9,7 @@
     "/info": {
       "get": {
         "summary": "Get netdata basic information",
-        "description": "The info endpoint returns basic information about netdata. It provides:\n* netdata version\n* netdata unique id\n* list of hosts mirrored (includes itself)\n* Operating System, Virtualization and Container technology information\n* List of active collector plugins and modules\n* number of alarms in the host\n  * number of alarms in normal state\n  * number of alarms in warning state\n  * number of alarms in critical state\n",
+        "description": "The info endpoint returns basic information about netdata. It provides:\n* netdata version\n* netdata unique id\n* list of hosts mirrored (includes itself)\n* Operating System, Virtualization, K8 node and Container technology information\n* List of active collector plugins and modules\n* number of alarms in the host\n  * number of alarms in normal state\n  * number of alarms in warning state\n  * number of alarms in critical state\n",
```
Comment (ilyam8) [inline web/api/netdata-swagger.json:12]:
```suggestion
        "description": "The info endpoint returns basic information about netdata. It provides:\n* netdata version\n* netdata unique id\n* list of hosts mirrored (includes itself)\n* Operating System, Virtualization, K8s node and Container technology information\n* List of active collector plugins and modules\n* number of alarms in the host\n  * number of alarms in normal state\n  * number of alarms in warning state\n  * number of alarms in critical state\n",
```

File: daemon/system-info.sh [hunk 2]
```diff
@@ -302,65 +302,76 @@ DISK_SIZE="unknown"
...
 
+# -------------------------------------------------------------------------------------------------
+# Detect whether the node is kubernetes node
+
+HOST_IS_K8S_NODE="false"
+
+if [ -n "${KUBERNETES_SERVICE_HOST}" ] && [ -n "${KUBERNETES_SERVICE_PORT}" ]; then
+  # These env vars are set for every container managed by k8s.
+  HOST_IS_K8S_NODE="true"
+elif pgrep "kubelet"; then
```
Comment (ilyam8) [inline daemon/system-info.sh:371]: This condition is true if i have k8s running in a container - minikube.

Minikube container is running - `_is_k8s_node: true`
Minikube container is stopped - `_is_k8s_node: false`

Perhaps we need to exclude Minikube (local testing) setups.

**Comment by vlvkobal** [comment]
- Don't we need this in anonymous statistics?
- `using-host-labels.md` should be changed.
- You should send the data to master (`sender.c`)
- You should change `netdata-swagger.yaml` as well.

**Comment by thiagoftsm** [comment]
>     * Don't we need this in anonymous statistics?

I think it is a good idea, and I added it.

>     * `using-host-labels.md` should be changed.
> 
>     * You should send the data to master (`sender.c`)
> 
>     * You should change `netdata-swagger.yaml` as well.

Everything was made, thanks to point them.

File: database/rrd.h [hunk 3]
```diff
@@ -719,6 +719,7 @@ struct rrdhost_system_info {
     char *virt_detection;
     char *container;
     char *container_detection;
+    char *is_k8_node;
```
Comment (ilyam8) [inline database/rrd.h:722]:
```diff
< is_k8_node
> is_k8s_node
```

File: docs/guides/using-host-labels.md [hunk 4]
```diff
@@ -50,6 +50,7 @@ read the status of your agent. For example, from a VPS system running Debian 10:
 {
   ...
   "host_labels": {
+    "_is_k8_node": "false",
```
Comment (ilyam8) [inline docs/guides/using-host-labels.md:53]:
```suggestion
    "_is_k8s_node": "false",
```

File: database/rrdhost.c [hunk 5]
```diff
@@ -982,6 +983,10 @@ static struct label *rrdhost_load_auto_labels(void)
         label_list =
             add_label_to_list(label_list, "_virt_detection", localhost->system_info->virt_detection, LABEL_SOURCE_AUTO);
 
+    if (localhost->system_info->is_k8s_node)
+        label_list =
+            add_label_to_list(label_list, "_is_k8_node", localhost->system_info->is_k8s_node, LABEL_SOURCE_AUTO);
```
Comment (ilyam8) [inline database/rrdhost.c:988]:
```suggestion
            add_label_to_list(label_list, "_is_k8s_node", localhost->system_info->is_k8s_node, LABEL_SOURCE_AUTO);
```

File: daemon/anonymous-statistics.sh.in [hunk 6]
```diff
@@ -63,6 +63,7 @@ if [ -n "$(command -v curl 2> /dev/null)" ]; then
     --data-urlencode "cd17=${NETDATA_CONTAINER_OS_VERSION}" \
     --data-urlencode "cd18=${NETDATA_CONTAINER_OS_VERSION_ID}" \
     --data-urlencode "cd19=${NETDATA_CONTAINER_OS_DETECTION}" \
+    --data-urlencode "cd20=${NETDATA_HOST_IS_K8S_NODE}" \
```
Comment (vlvkobal) [inline daemon/anonymous-statistics.sh.in:66]: You should add it to wget as well.

**Comment by vlvkobal** [comment]
How come the changes to the content of swagger files are different? You should edit API documentation in Swagger Editor. To save `netdata-swagger.yaml` and `netdata-swagger.json` use `File`->`Save as YAML` and `File`->`Convert and save as JSON`.

**Comment by vlvkobal** [comment]
You should add to the API documentation something like this:
https://github.com/netdata/netdata/blob/master/web/api/netdata-swagger.yaml#L967-L970

**Comment by ilyam8** [comment]
@thiagoftsm it seems you ran some auto format tool in the last commit?

**Comment by thiagoftsm** [comment]
> @thiagoftsm it seems you ran some auto format tool in the last commit?

The last commit has files exported direct from swagger editor. I will take a look on these warnings.

**Comment by thiagoftsm** [comment]
> @thiagoftsm it seems you ran some auto format tool in the last commit?

I export again using other source, and it kept the previous format.

**Review by vlvkobal — APPROVED** [review]

**Review by thiagoftsm — APPROVED** [review]

**Pull request merged**